bigchaindb/core.py

```diff
@@ -433,14 +387,17 @@ def get_owned_ids(self, owner):
         """
         return self.get_outputs_filtered(owner, include_spent=False)
 
+    @property
+    def fastquery(self):
+        return fastquery.FastQuery(self.connection, self.me)
+
     def get_outputs_filtered(self, owner, include_spent=True):
```

Comment:
I had the suspicion that `get_outputs` or `get_outputs_filtered` would be used in the consensus logic.

I checked through the code, but neither:

- `get_spent`
- `get_owned_ids`
- `get_transaction`

uses `get_output_filtered` when validating. Meaning that `get_outputs_filtered` does not influence the consensus logic. Good.


However, this leads me to the question of: How can we make sure that a "unsafe" method like `get_outputs_filtered` will never get used by any collaborator accidentally (e.g. because they don't know that it's not checking the block's election status).

A few options come to mind:

1. Have `get_outputs_filtered` return results based on the federation's block election status by default (only use the unsafe method in the HTTP API with a big warning.
  - Add a parameter `safe=True|False` to the method?
2. Put a big warning comment above the method explaining that it should never be used in consensus functions (however: what about e.g. `get_owned_ids`, which is a proxy of `get_outputs_filtered`? It should also never be used
3. ...?

Anyone have best practices for this?